qt/scientific_interfaces/Inelastic/Processor/ElwinPresenter.cpp

```diff
@@ -140,12 +140,13 @@ void ElwinPresenter::handleValueChanged(std::string const &propName, bool value)
  */
 void ElwinPresenter::newInputDataFromDialog() {
   // Clear the existing list of files
-  m_view->clearPreviewFile();
-  m_view->newInputDataFromDialog(m_dataModel->getWorkspaceNames());
+  m_view->updatePreviewWorkspaceNames(m_dataModel->getWorkspaceNames());
 
-  std::string const wsname = m_view->getPreviewWorkspaceName(0);
-  auto const inputWs = WorkspaceUtils::getADSWorkspace(wsname);
-  setInputWorkspace(inputWs);
+  // sets last added workspace as input workspace
+  auto const lastAddedWSIndex = m_dataModel->getNumberOfWorkspaces() - 1;
+  m_view->setPreviewWorkspaceName(static_cast<int>(lastAddedWSIndex.value));
+  auto lastAddedWS = m_dataModel->getWorkspace(lastAddedWSIndex);
+  setInputWorkspace(lastAddedWS);
```

Comment:
This is the section that I think you could remove, because its a bit confusing and I think it could be handled anyway by signals and slots. See my comment below